test/unit.test.js

```diff
@@ -90,7 +90,11 @@ for (const { testName, isRoot } of unitTests) {
     console.log(`Skipping unit test on Node.js 22 or above: ${testSuffix}`);
     continue;
   }
-  const unitPath = join(__dirname, 'unit', testName);
+  let unitPath = join(__dirname, 'unit', testName);
+
+  if (testName === 'fs-relative-emission') {
+    unitPath += '/main';
```

Comment:
```suggestion
    // this test has file reads that traverse to the parent dir
    // so we need to start from the child dir.
    unitPath += '/main';
```

Lets mention why this is here